File: Data Structures & Algorithms/two-integer-sum/submission-2.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        int[] solution = new int[2];
        Map<Integer, Integer> map = new HashMap<>();
        for(int i = 0; i < nums.length; i++){
            int difference = target - nums[i];
            if(map.containsKey(difference)){
                int val = map.get(difference);
                if(val>i){
                    
                    solution[0] = i;
                    solution[1] = val;
                    return solution;
                } else {
                    solution[0] = val;
                    solution[1] = i;
                    return solution;
                }
            } else {
                map.put(nums[i],i);
            }
        }
        return solution;
    }
}

File: Data Structures & Algorithms/subarray-sum-equals-k/submission-2.py
class Solution:
    def subarraySum(self, nums: List[int], k: int) -> int:
        #Use the hashamp to see if there is a chopped off bit in there.  
        res = 0
        hashmap = {0 : 1}
        prefixSum = 0
        for num in nums:
            prefixSum += num
            print(prefixSum, num)
            target = prefixSum - k
            if target in hashmap:
                res += hashmap[target]
            hashmap[prefixSum] = hashmap.get(prefixSum, 0) + 1
            print(res)
        return res
